src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7161,50 +7161,20 @@ private FixedStatementSyntax ParseFixedStatement()
             _termState = saveTerm;
 
             var closeParen = this.EatToken(SyntaxKind.CloseParenToken);
-            StatementSyntax statement = this.ParseEmbeddedStatement();
+            StatementSyntax statement = this.ParseStatementCore() ?? ParseEmptyStatement();
             return _syntaxFactory.FixedStatement(@fixed, openParen, decl, closeParen, statement);
         }
 
+        private EmptyStatementSyntax ParseEmptyStatement()
+            => SyntaxFactory.EmptyStatement(EatToken(SyntaxKind.SemicolonToken));
+
         private bool IsEndOfFixedStatement()
         {
             return this.CurrentToken.Kind == SyntaxKind.CloseParenToken
                 || this.CurrentToken.Kind == SyntaxKind.OpenBraceToken
                 || this.CurrentToken.Kind == SyntaxKind.SemicolonToken;
         }
 
-        private StatementSyntax ParseEmbeddedStatement()
```

Comment:
>private StatementSyntax ParseEmbeddedStatement() [](start = 8, length = 48)

We should just reduce the size of `ParseEmbeddedStatement()` rather than inlining the method in each caller. 

For instance, we could remove the locals by extracting the "stand-alone expression statements" part to a local function.

```C#
private StatementSyntax ParseEmbeddedStatement()
{
    return handleStandaloneStatement(
        this.ParseStatementCore() ??
        SyntaxFactory.EmptyStatement(EatToken(SyntaxKind.SemicolonToken)));

    StatementSyntax handleStandaloneStatement(StatementSyntax statement)
    {
        switch (statement.Kind)
        {
            // ...
        }
    }
}
```

Reply:
won't this still produce a whole new stack frame?

Reply:
> won't this still produce a whole new stack frame?

For the local function? Yes, but only as the last frame in a recursive call stack.

Reply:
Can you clarify that.  Why would it not show up over and over again.  i.e. `ParseEmbeddedStatement -> ParseStatementCore -> ParseIfStatement -> ParseEmbeddedStatement -> ...`

?

My goal here is to dramatically reduce the cost of statement parsing by just fully removing these unnecessary intermediary frames.  i.e. instead of if statement parsin needing 3 frames per recursion, it should only take 2.  Don't stack frames have overhead (even with no locals)?  I think this should be one of the largest ways to help out the deeply-nested scenario, right?

Reply:
> Why would it not show up over and over again.

The local function is only called after the nested `ParseStatementCore()` has returned.

> My goal here is to dramatically reduce the cost of statement parsing by just fully removing these unnecessary intermediary frames.

We should try simplifying the code first and only make the code more complicated if simpler alternatives are not sufficient. If the methods are simple enough, the JIT may inline the calls.

Reply:
Ok.  I'll try this and will let you know the results.

Reply:
Yup.  Can keep the original method intact.  it looks like it does get inlined.

Reply:
Note: if i use the `handleStandaloneStatement` approach, then i lose 60 stack frames.  It doesn't seem worth it.  I'm going to keep the check in the binder